src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -102,8 +102,11 @@ class MulticopterPositionControl : public ModuleBase<MulticopterPositionControl>
 	int print_status() override;
 
 private:
-
-	bool 		_in_smooth_takeoff = false; 		/**< true if takeoff ramp is applied */
+	// smooth takeoff vertical velocity ramp state
+	bool _in_takeoff_ramp = false; /**< true while takeoff ramp is applied */
+	bool _position_triggered_takeoff = false; /**< true if takeoff was triggered by a position setpoint */
+	float _takeoff_ramp_velocity = -1.f; /**< current value of the smooth takeoff ramp */
```

Comment:
Just a note and not more: to me `velocity` agrees with the coordinate convention (vs speed, which is always positive). Hence, with that convention in mind, `_takeoff_ramp_velocity` of `-1` means that it would start with 1m/s upward.

Reply:
Good point, I should have switched to the proper coordinate convention. Noted to do in the next iteration since this is a release blocker. 👍

Reply:
When #12014 is finished the coordinate frame will be correct.